PLUGINS.md

```diff
@@ -0,0 +1,180 @@
+    |   |   └── ...
+    │   └── ...
+    └── ...
+```
+
+## Plugin Directory
+
+Plugin directory name should match the plugin name (console plugins are an exception), as a convention plugin 
+directory names should end with the `-plugin` suffix (eg. `my-plugin`).
+
+- Use lowercase dash-separated names for all files (to avoid git issues with case-insensitive file systems)
+- Exceptions are files which have their own naming conventions (eg Makefile, README)
+
+## Package.json
+
+Plugins are packages, the package definitions of a plugin should include a `consolePlugin`
+map that defines the plugin declarations file, and the plugin integration tests.
+
+``` json
+// File: package.json
+
+{
+  "name": "@console/demo-plugin",
+  "version": "0.0.0-fixed",
+  "description": "Demo plugin for Console web application",
+  "private": true,
+  "dependencies": {
+    "@console/plugin-sdk": "0.0.0-fixed",
+    "@console/shared": "0.0.0-fixed"
+  },
+  "devDependencies": {
+    "@console/internal-integration-tests": "0.0.0-fixed"
+  },
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    "integrationTests": {
+      "demo": [
+        "@console/internal-integration-tests/tests/base.scenario.ts",
+        "integration-tests/tests/demo.scenario.ts"
+      ]
+    }
+  }
+}
+```
+
+## Plugin Decelerations File (`.../src/plugin.tsx`)
+
+The plugin declaration entry file is defined in `package.json` in the `consolePlugin: entry` variable,
+relative to the plugin root directory.
+
+``` json
+// File: package.json
+
+{
+  ...
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    ...
+  }
+}
+```
+
+Plugin declaration file define the plugin extensions to the backend resources and to the UI components.
+
+It will export an array of plugin objects (type `Plugin<ConsumedExtensions>`) each `Plugin` object have a `type`
+declaring the plugin capabilities and options (for example `Page/Route`, `Dashboards/Card` types). 
+
+``` es6
+// File: src/plugin.tsx
+
+const plugin: Plugin<ConsumedExtensions> = [
+  ...
+  {
+    type: 'Dashboards/Tab',
+    properties: {
+      id: 'persistent-storage',
+      title: 'Persistent Storage',
+    },
+  },
+  ...
+];
+
+export default plugin;
+```
+
+## Unit Tests
+
+Unit tests use [jest](https://jestjs.io/) and run automatically from files in any directory named `.../__tests__` 
+and ending with `.spec.(ts|tsx|js|jsx)` suffix.
+
+``` bash
+  .
+  ├── my-plugin
+  |   ├── src
+  |   |   ├── __tests__
+  |   |   |   ├── some-test.spec.ts
+  |   |   |   └── ...
+  |   |   └── ...
+  |   └── ...
+  └── ...
+```
+
+## Integration Tests
+
+Integration tests use [protractor](http://www.protractortest.org) and must be defined in the `consolePlugin` map, in the plugins `package.json`.
+Test path is relative the plugin root directory. It is possible to include common tests defined in `@console/internal-integration-tests`,
+and use common testing methods defined in the internal package by requiring it in the test.
```

Comment:
"Test path is relative the plugin root" -> "The test path is relative to the plugin root"

You can delete the comma after "`@console/internal-integration-tests`".